exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/sequencefile/SequenceFileRecordReader.java

```diff
@@ -0,0 +1,146 @@
+
+import java.io.IOException;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.concurrent.TimeUnit;
+
+import com.google.common.base.Stopwatch;
+import org.apache.drill.common.exceptions.DrillRuntimeException;
+import org.apache.drill.common.exceptions.ExecutionSetupException;
+import org.apache.drill.common.expression.SchemaPath;
+import org.apache.drill.common.types.TypeProtos;
+import org.apache.drill.common.types.Types;
+import org.apache.drill.exec.exception.SchemaChangeException;
+import org.apache.drill.exec.ops.OperatorContext;
+import org.apache.drill.exec.physical.impl.OutputMutator;
+import org.apache.drill.exec.record.MaterializedField;
+import org.apache.drill.exec.store.AbstractRecordReader;
+import org.apache.drill.common.types.TypeProtos.MajorType;
+import org.apache.drill.exec.vector.NullableVarBinaryVector;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.BytesWritable;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.hadoop.mapred.FileSplit;
+import org.apache.hadoop.mapred.Reporter;
+import org.apache.hadoop.mapred.SequenceFileAsBinaryInputFormat;
+
+
+public class SequenceFileRecordReader extends AbstractRecordReader {
+  private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(SequenceFileRecordReader.class);
+
+  private static final int PER_BATCH_RECORD_COUNT = 4096;
+  private static final int PER_BATCH_BYTES = 256*1024;
+
+  private static final MajorType KEY_TYPE = Types.optional(TypeProtos.MinorType.VARBINARY);
+  private static final MajorType VALUE_TYPE = Types.optional(TypeProtos.MinorType.VARBINARY);
+
+  private final SchemaPath keySchema = SchemaPath.getSimplePath("binary_key");
+  private final SchemaPath valueSchema = SchemaPath.getSimplePath("binary_value");
+
+  private NullableVarBinaryVector keyVector;
+  private NullableVarBinaryVector valueVector;
+  private final FileSplit split;
+  private org.apache.hadoop.mapred.RecordReader<BytesWritable, BytesWritable> reader;
+  private final BytesWritable key = new BytesWritable();
+  private final BytesWritable value = new BytesWritable();
+  private final Configuration fsConf;
+
+  public SequenceFileRecordReader(final FileSplit split,
+                                  final Configuration fsConf) {
+    final List<SchemaPath> columns = new ArrayList();
+    columns.add(keySchema);
+    columns.add(valueSchema);
+    setColumns(columns);
+    this.fsConf = fsConf;
+    this.split = split;
+  }
+
+  @Override
+  protected boolean isSkipQuery() {
+    return false;
+  }
+
+  @Override
+  public void setup(OperatorContext context, OutputMutator output) throws ExecutionSetupException {
+    final SequenceFileAsBinaryInputFormat inputFormat = new SequenceFileAsBinaryInputFormat();
+    final JobConf jobConf = new JobConf(fsConf);
+    jobConf.setInputFormat(inputFormat.getClass());
+    try {
+      this.reader = inputFormat.getRecordReader(split, jobConf, Reporter.NULL);
```

Comment:
Just saw sudheesh comment about moving things back to the constructor... We shouldn't have anything at all in the constructor that doesn't need to be there. No work should be done. Even opening files or seeking within them. That should be done in setup because we may have 1000 record readers that need to be constructed in a single thread where we then have a limit 5. We want to only run setup on the first one and then terminate. No reason to do any setup work on anything that we later realize is not necessary.